client/client.go

```diff
@@ -79,58 +93,68 @@ func GetRawQuote(d Device, reportData [64]byte) ([]uint8, uint32, error) {
 	}
 	result, err := d.Ioctl(labi.IocTdxGetQuote, &tdxReq)
 	if err != nil {
-		return nil, 0, err
+		return nil, err
 	}
 	if result != uintptr(labi.TdxAttestSuccess) {
-		return nil, 0, fmt.Errorf("unable to get the quote")
+		return nil, fmt.Errorf("unable to get the quote")
 	}
 	if tdxHdr.Status != 0 {
 		if labi.GetQuoteInFlight == tdxHdr.Status {
-			return nil, 0, fmt.Errorf("the device driver return busy")
+			return nil, fmt.Errorf("the device driver return busy")
 		} else if labi.GetQuoteServiceUnavailable == tdxHdr.Status {
-			return nil, 0, fmt.Errorf("request feature is not supported")
+			return nil, fmt.Errorf("request feature is not supported")
 		} else if tdxHdr.OutLen == 0 || tdxHdr.OutLen > labi.ReqBufSize {
-			return nil, 0, fmt.Errorf("invalid Quote size: %v. It must be > 0 and <= : %v", tdxHdr.OutLen, labi.ReqBufSize)
+			return nil, fmt.Errorf("invalid Quote size: %v. It must be > 0 and <= : %v", tdxHdr.OutLen, labi.ReqBufSize)
 		}
 
-		return nil, 0, fmt.Errorf("unexpected error: %v", tdxHdr.Status)
+		return nil, fmt.Errorf("unexpected error: %v", tdxHdr.Status)
 	}
 
-	return tdxHdr.Data[:tdxHdr.OutLen], tdxHdr.OutLen, nil
+	return tdxHdr.Data[:tdxHdr.OutLen], nil
 }
 
-// GetQuote call GetRawQuote to get the quote in byte array and convert it into proto.
-func GetQuote(d Device, reportData [64]byte) (*pb.QuoteV4, error) {
-	quotebytes, size, err := GetRawQuote(d, reportData)
-	if err != nil {
-		return nil, err
-	}
-	if len(quotebytes) > int(size) {
-		quotebytes = quotebytes[:size]
+// getRawQuoteViaProvider use QuoteProvider to fetch quote in byte array format.
+func getRawQuoteViaProvider(qp QuoteProvider, reportData [64]byte) ([]uint8, error) {
+	if err := qp.IsSupported(); err == nil {
+		quote, err := qp.GetRawQuote(reportData)
+		return quote, err
 	}
-	return convertRawQuoteToProto(quotebytes)
+	return fallbackToDeviceForRawQuote(reportData)
 }
 
-// GetRawQuoteViaProvider use QuoteProvider to fetch quote in byte array format.
-func GetRawQuoteViaProvider(qp QuoteProvider, reportData [64]byte) ([]uint8, error) {
-	if err := qp.IsSupported(); err == nil {
-		return qp.GetRawQuote(reportData)
+// GetQuote uses Quote provider or Device(deprecated) to get the quote in byte array and convert it
+// into proto.
+// Supported quote formats - QuoteV4.
+func GetQuote(quoteProvider any, reportData [64]byte) (any, error) {
+	switch qp := quoteProvider.(type) {
+	case Device:
+		return getQuoteViaDevice(qp, reportData)
+	case QuoteProvider:
+		return getQuoteViaProvider(qp, reportData)
```

Comment:
This function can directly call GetRawQuote() and then convert the byte array to proto. In this case you wouldn't need extra functions(getQuoteViaDevice, getQuoteViaProvider)

Reply:
Done